nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -114,6 +126,83 @@ public static DocumentModel retrieveContentFromColdStorage(CoreSession session,
+     *            {@value COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME}
+     */
+    public static ColdStorageContentStatus checkAvailabilityOfColdStorageContent(CoreSession session) {
+        log.debug("Start checking the available cold storage content which are being retrieved for repository: {}",
+                session::getRepositoryName);
+
+        // as the volume of result will be small, we don't use BAF
+        DocumentModelList documents = session.query(GET_DOCUMENTS_TO_CHECK_QUERY);
+
+        // for every available content we will fire an event
+        int available = 0;
+        EventService eventService = Framework.getService(EventService.class);
+        for (DocumentModel doc : documents) {
+            if (isColdStorageContentAvailable(doc)) {
+                available++;
+                DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+                Event event = ctx.newEvent(COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME);
+                eventService.fireEvent(event);
+            }
+        }
+
+        int total = documents.size();
+        log.debug("End checking the available cold storage content for repository: {}, found: {}/{}",
+                session.getRepositoryName(), available, total);
+
+        return new ColdStorageContentStatus(session.getRepositoryName(), total, available);
+    }
+
+    /**
+     * Checks if the retrieved cold storage content associated with the given document is available for downloading.
+     * {@link #retrieveContentFromColdStorage(CoreSession, DocumentRef, int)}
+     * {@link #checkAvailabilityOfColdStorageContent(CoreSession)}
+     */
+    public static boolean isColdStorageContentAvailable(DocumentModel documentModel) {
+        // FIXME to be reworked depending on how we will check the availability on low level
+        return documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) != null
+                && documentModel.hasFacet(FacetNames.COLD_STORAGE);
+    }
+
+    /**
+     * Gives status about the cold storage content being retrieved or are available for a given repository.
+     */
+    public static class ColdStorageContentStatus {
+
+        protected final String repositoryName;
+
+        protected final int totalBeingRetrieved;
+
+        protected final int totalContentAvailable;
+
+        public ColdStorageContentStatus(final String repositoryName, final int totalBeingRetrieved,
+                final int totalContentAvailable) {
+            this.repositoryName = repositoryName;
+            this.totalBeingRetrieved = totalBeingRetrieved;
+            this.totalContentAvailable = totalContentAvailable;
+        }
+
+        public String getRepositoryName() {
+            return repositoryName;
+        }
+
+        public int getTotalBeingRetrieved() {
+            return totalBeingRetrieved;
+        }
+
+        public int getTotalContentAvailable() {
+            return totalContentAvailable;
```

Comment:
Do we need these for now? What's their purpose?

Reply:
for now the only use is in the test but my idea is to have the status of scheduled job available for more logging ...